test-case/check-capture.sh

```diff
@@ -61,7 +61,7 @@ file_prefix=${OPT_VALUE_lst['f']}
 
 func_lib_setup_kernel_last_line
 func_lib_check_sudo
-func_pipeline_export $tplg "type:capture & ${OPT_VALUE_lst['S']}"
+func_pipeline_export $tplg "type:any & ${OPT_VALUE_lst['S']}"
```

Comment:
Why not relay on the `func_pipeline_export` to do the type filter? I think make check to each end user is not a good idea. I think this would be a degenerate for our function.

Reply:
The pipeline filtering feature is provided by sof-tplgreader.py, and the filtering feature is not implemented for sof-dump-status.py.  If we need a common filtering feature shared between both, it requires a lot of code change, because the filter code in the sof-tplgreader.py is in a class, and requires `self`, which need some efforts to refactor.

This maybe done later

Reply:
I checked the output from, I think add a simple filter for playback/capture will be very easy. (a simple grep will make it work)
```
cat /proc/asound/pcm
00-00: Port5 (*) :  : playback 1 : capture 1
00-01: DMIC (*) :  : capture 1
00-02: DMIC16kHz (*) :  : capture 1
00-04: Media Playback 4 (*) :  : playback 1
00-05: HDMI1 (*) :  : playback 1
00-06: HDMI2 (*) :  : playback 1
00-07: HDMI3 (*) :  : playback 1
```

Reply:
Assuming the format of `/proc/asound/pcm` is stable and guaranteed then a simple `grep` in `func_pipeline_export()` seems like a cheap and good idea to me.

> PS: to me a patch that influence so many files is also a big risk. 

There is 1 new file which is just extracted from an existing one and only 4 other modified files. One of the modified file is just getting a one-line function.

> Changing API is a dangerous thing and need more test. So I would prefer to keep old ABI 

What specific API and ABI are changing here?

> and add simple implement for the legacy HDA.

I don't get this sorry, can you elaborate?